code/modules/atmospherics/machinery/components/unary_devices/vent_pump.dm

```diff
@@ -154,6 +238,17 @@
 	var/turf/open/us = loc
 	if(!istype(us))
 		return
+
+	var/current_integrity = get_integrity()
+	if(fan_overclocked)
+		current_integrity = update_integrity(current_integrity - fan_damage_rate)
+		if(current_integrity == 0)
```

Comment:
You should handle this in atom_break, and 0 integrity is usually used to indicate complete destruction so you may want to either increase integrity a bit more and set a higher level for break, or reduce the damage per tick for the equivalent.